### ISSUE TITLE
Option `allowOutsideClick` should accept a boolean

### ISSUE THREAD

**Issue by SeanMcP** [focus-trap/focus-trap-react#86]
## Description

The prop type for `allowOutsideClick` is `func`:

https://github.com/focus-trap/focus-trap-react/blob/749f02180e8c7cb6c237723419fbb81e739c261a/src/focus-trap-react.js#L143

This is fine (and [mimics the `focus-trap` options](https://github.com/davidtheclark/focus-trap#focustrap--createfocustrapelement-createoptions)), but the name implies that it should also accept a boolean. "allows" is one of those words – like "has", "is", and "will" – that indicates to the user that the value is either `true` or `false`.

## Expected behavior

Passing `true` to the `allowOutsideClick` works as expected.

## Actual behavior

A run-time warning is logged and the option is not active.

## Possible solution

In this loop: https://github.com/focus-trap/focus-trap-react/blob/749f02180e8c7cb6c237723419fbb81e739c261a/src/focus-trap-react.js#L31-L47

We could add a type check like:

```js
if (optionName === 'allowOutsideClick' && typeof specifiedFocusTrapOptions[optionName] === 'boolean') {
    tailoredFocusTrapOptions[optionName] = () => specifiedFocusTrapOptions[optionName]
}
```

Then update the prop types at the bottom to be:

```js
allowOutsideClick: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.bool
]),
```

## Conclusion

This is a small patch that would make this little feature a little more usable. I'd be happy to contribute and make this addition, but yield to the maintainers for further discussion.

**Comment by maraisr** [comment]
Hi @SeanMcP thanks for the issue! 

I only recently took over maintainer role, so I'd need to look into this more closely. Ie, to see why this was a function.

At face value I think this wouldn't be such a bad idea, and probably more performant too 🤷‍♂️ so yeah, if you wanna raise a PR, I'd be more than happy to take a look! :)

Thanks again for taking the time for a well-formed issue ❤️

**Comment by SeanMcP** [comment]
@maraisr I'll put a PR together later today 👍

**Comment by stefcameron** [comment]
@SeanMcP Thanks for raising this issue and for the details in it! I'm sorry I only just noticed this now. Like @maraisr, I just took over as maintainer recently too.

So I'm looking at this, and at the option on `focus-trap` itself. I agree the name implies a boolean, so I like your idea of supporting both a boolean and a function, and I like your suggestion for how to address it.

However, `allowOutsideClick` has no special handling in `focus-trap-react`: It just passes it down to `focus-trap`, and I feel like if we do this here, we should really be doing it in `focus-trap` instead. Then the only fix/update needed here would be to change the prop-type to be `oneOfType([bool, func])`.

If we do this here, I'm concerned this library would start to diverge and that could get us into hot water down the road. `focus-trap-react` is really meant to "bring" or "make it easier to use" `focus-trap` in a React app.

Thoughts? Would you be open to doing something similar to #87 but on `focus-trap`? (Though first, @maraisr and I would have to get that repo moved into this new org...)

**Comment by SeanMcP** [comment]
@stefcameron I think that is a good idea. While we could make the change here without any big fear of the API's drifting apart, the change really belongs in `focus-trap`. I'd be happy to open another PR, but should I wait until the repo is transferred to this organization? I saw David 👍 your suggestion, so just let me know what the order of operations that you would prefer.

**Comment by stefcameron** [comment]
Cool! Thanks, @SeanMcP!

**Comment by stefcameron** [comment]
Ideally, we'd add the feature in `focus-trap`, and then to fix this issue, we'd just keep the `prop-types` part of #87 and remove the rest.

**Comment by stefcameron** [comment]
This was fixed with focus-trap/focus-trap#120 and #87.
